packages/main/src/components/AnalyticalTable/index.tsx

```diff
@@ -655,6 +656,105 @@ const AnalyticalTable = forwardRef<AnalyticalTableDomRef, AnalyticalTablePropTyp
     );
   };
 
+  const handleOnAutoResize = (e, accessor) => {
```

Comment:
I have been working on refactoring this as a hook. It now follows the pattern of useDynamicColumnWidths and canvas is removed.

I do need some information from the ColumnHeaderContainer that I cannot get from the instance (or maybe there is a way to tap into the information I need, and I'm missing it). 

My first attempt is to return a function to be fed by the onDoubleClick method of the data-resizer. Like useResizeColumnsConfig, it pushes a click handler (onDoubleClick) into the getResizerProps hook provided. 

Unlike, useResizeColumnsConfig, a local onDoubleClick in ColumnHeaderContainer feeds the necessary information to the hook using column.getResizerProps().onDoubleClick. 

I am not sure how I feel about this in general. getResizerProps seems like the right place to put autoResize, but I also receive information I do not need and do not use.

I am also conflicted on detecting the button/icon. I decided to create an ID in Expandable that can be searched for and measured. Otherwise, it might be possible to do this by using properties and themes to decide which type of button/icon is being displayed. This option felt less future proof.

This does not work on TreeTables and tables with strings in the groupBy array (noted in documentation). 

This commit outdates other comments, so I'm leaving them unanswered for now. 

Thanks for your help. Looking forward to further review.